output/baseoutput/clientsession.go

```diff
@@ -217,17 +231,18 @@ func (session *clientSession) collectLeftovers(maybePreviousLeftovers chan base.
 		session.logger.Panic("invalid ending type: ", ending)
 	}
 
-	if !session.ackerEnded.Wait(defs.ForwarderAckerStopTimeout) {
-		session.logger.Errorf("BUG: timeout waiting for acknowledger to stop. stack=%s", util.Stack())
+	if !session.ackerEnded.Wait(defs.IntermediateChannelTimeout) {
+		session.logger.Errorf("BUG: timeout waiting for acknowledger to hard stop. stack=%s", util.Stack())
 	}
 	fromAckerChannel := collectChunksFromChannel(session.ackerChan)
 
 	// gather pendings chunks left in runAcknowledger's pendingChunksByID
 	var fromAckerPending []base.LogChunk
-	if pendingListPtr := session.unacked.Load().(*[]base.LogChunk); pendingListPtr != nil {
-		fromAckerPending = *pendingListPtr
+	pendingListPtr, ok := session.unacked.Load().(*[]base.LogChunk)
+	if !ok || pendingListPtr == nil {
+		session.logger.Errorf("BUG: failed to get un-ACK'ed chunks from acknowledger. ok=%v, pendingListPtr=%v, stack=%s", ok, pendingListPtr, util.Stack())
 	} else {
-		session.logger.Errorf("BUG: failed to get un-ACK'ed chunks from acknowledger. stack=%s", util.Stack())
```

Comment:
On [line 265](https://github.com/relex/slog-agent/blob/fix/session-close-bug/output/baseoutput/clientsession.go#L265) we log the amount of pending chunks received here, so I don't think there's any need for a duplicate log.

Reply:
Logging in baseoutput is a lot of more verbose than other parts because initialization and finalization that could go wrong very easily. We could log in debug level (invisible by default) if it'd help anything.